### ISSUE TITLE
Add reading mode to break even if idle detected

### ISSUE THREAD

**Issue by AllanChain** [AllanChain/sane-break#11]
See also #3 

Sometimes the user want to enable break even if idle is detected. For example:
- The user prefers to take regular breaks when watching video
- The user is reading (mostly idle, with occasional activities)
- The user is writing on paper while referring to information displayed on the screen.

Therefore, It can be helpful to have a reading mode in which break will happen even if idle detected.

However, considering this mode is not frequently used, and the user may not always remember turning on this mode manually, we put this feature to medium priority.

**Comment by wolftune** [comment]
I thought maybe this be like "postpone" and the UI be in the context menu, and user has to choose a time-length for reading mode, and after that time it automatically returns to normal mode… but one of the options could be "indefinite" or similar (no time-limit, have to manually turn off reading mode)

But since reading-mode means *more* breaks, I think it's less of a risk that user forgets to turn it off when not wanted, so maybe this time-for-reading idea is not needed.

**Comment by Erudition** [comment]
Preplanning the reading duration notwithstanding -- the downside is sane-break can't tell when you've _stopped_ reading, and actually stepped away.

This is the same with videos. Normally, with video watching it's annoying to have to touch an input device every once in a while, if it were necessary to keep the device from sleeping or screen from turning off, for example. Mostly because it means revealing the video controls HUD overlay (especially for full-screen watching) and then waiting for it to go away.

But to me, that's not necessarily the case with reading! To scroll or switch pages when actively reading, you'll be occasionally triggering some input anyway -- and if you're reading particularly slowly, an occasional touch of the mouse doesn't get much in the way.

All of this is to mention that the "safest" workaround (most accurate tracking, screen only unlocked when present, etc) I can think of without sophisticated presence sensing (e.g. webcam face detection) is to set one's screen timeout to a low duration, like 1, 3 or 5 minutes. We're already used to instinctively touching the mouse to avoid the screen falling asleep, and this input would then serve a double purpose to indicate to sane-break we're still active.

Best part is, if you're not there, your screen will save power, your desktop will be locked from other's eyes (if desired), and the auto break tracking will only ever be off by at most 3 minutes, for example.

**Comment by wolftune** [comment]
But reading-mode (which incidentally was brought up because WorkRave has it) means the breaks are prompted no-matter the activity.

It's a mode that just means the breaks come on regular timing with no delays. And the transition from nag to full-screen should happen in a fixed number of seconds (perhaps settable). Activity is irrelevant in this mode.

Thus, there's not really a risk of accidentally leaving it on. When you are done reading and back to normal mode, the fact that it will *ignore* your activity and break no matter what is potentially enough for the user to think "oh, turn off reading mode". The important thing is that reading-mode isn't a method to *avoid* breaks, it makes them happen *more*. So, there isn't a risk of reading mode being left on and leading to missed breaks.

**Comment by AllanChain** [comment]
I have received some bug reports about reading mode that make the app stop working. For example, the timer is not ticking or the app freezes. I have no idea why this happens, and there are different problems with different devices. I have to remove this feature to make the app stable.

**Comment by Erudition** [comment]
@wolftune You're right, I misunderstood the feature. I guess I don't have a use for it - if I take a break naturally, I want it to count, regardless of what I was doing on the screen. Mostly because I don't want to come back from a break just to have it nag me for a break which I then need to postpone (or skip, in the case of WorkRave, which would make the stats inaccurate).

**Comment by wolftune** [comment]
@Erudition the idea as it works in Work-Rave is:

- manually turn on reading mode
- probably make a clear indication on the icon that it is on
- manually turn it off when done with extended-reading-time

Because it is *annoying* indeed to come back from a natural break and still get nagged:

- use middle-click to start a break immediately when naturally breaking
- you will not otherwise forget to turn off reading mode because it is an annoying mode if you aren't needing it

So, again, the point is: if someone is reading some long document slowly and is mostly failing to take breaks because Sane-Break thinks they are inactive and isn't reminding them, then they are happy to have this mode. I think most people won't use it, and those that do will use it only rarely.
